### PR TITLE
ENH: Add FunctionalComp.gradient.derivative

### PR DESCRIPTION
**Pull request by adler-j**
Simple implementation for when it exists and we can "easily" find it.

### CODE REVIEW

File: odl/solvers/functional/functional.py [hunk 1]
```diff
@@ -576,6 +576,17 @@ def _call(self, x):
...
                 return op.derivative(x).adjoint(func.gradient(op(x)))
 
+            def derivative(self, x):
+                """The derivative in point ``x``.
+
+                This is only defined
+                """
+                if not op.is_linear:
+                    raise NotImplementedError('derivative only implemented '
+                                              'linear opertors.')
```
Comment (kohr-h) [inline odl/solvers/functional/functional.py]: "for" missing

**Comment by adler-j** [comment]
Fixed comment, will merge after CI.

**Comment by adler-j** [comment]
It seems I wont. Any idea what's up with the error message?

**Comment by adler-j** [comment]
I've reproduced it locally, it is because travis upgraded scipy to 0.19, which apparently has a breaking change here, related to `future.newint`, see this issue at numpy: https://github.com/numpy/numpy/issues/8787

**Pull request merged**